### PR TITLE
Async HttpClient builders to support custom IOSessionListener 

### PR DESCRIPTION
**Pull request by cda007**
Add the option to set an IOSessionListener on HttpAsyncClientBuilder

### CODE REVIEW

**Comment by SmallerCoder** [comment]
邮件已收到！我会尽快阅读！谢谢！

File: httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/HttpAsyncClientBuilder.java [hunk 1]
```diff
@@ -318,7 +320,15 @@ public final HttpAsyncClientBuilder setIOReactorConfig(final IOReactorConfig ioR
         this.ioReactorConfig = ioReactorConfig;
         return this;
     }
-
```
Comment (ok2c) [inline httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/HttpAsyncClientBuilder.java:321]: @cda007 Could you please add `@since 5.2` tag?

**Review by ok2c — CHANGES_REQUESTED** [review]
@cda007 Could you please also make symmetric changes to `H2AsyncClientBuilder`?

File: httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/HttpAsyncClientBuilder.java (hunk 1 above)
Comment (cda007) [reply httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/HttpAsyncClientBuilder.java:321]: done

**Review by ok2c — APPROVED** [review]

**Comment by cda007** [comment]
Hi Oleg,

Sorry for this. As I told you I am not familiar with this process and with git. Is there any specific pre-commit hook that would avoid to commit code not respecting rules ?

Regards,
Christophe

**Comment by ok2c** [comment]
@cda007 This has nothing to do with git. This is straight Maven. `mvn clean verify` has got to pass for you locally with your changes. Style rules like no trailing blanks are enforced by the Maven style-check plugin.

File: httpclient5/src/main/java/org/apache/hc/client5/http/CircularRedirectException.java [hunk 2]
```diff
@@ -39,7 +39,7 @@
      * Creates a new CircularRedirectException with a {@code null} detail message.
      */
     public CircularRedirectException() {
-        super();
+        super();
```
Comment (michael-o) [inline httpclient5/src/main/java/org/apache/hc/client5/http/CircularRedirectException.java:42]: Redundant space

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by cda007** [comment]
Fixed. I use now to maven verification tool, should not occur anymore

**Pull request merged**